test/datatype/check_op.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
```

Comment:
I dont want this test to fail on the first failure. I want to see all failures.

Reply:
The `make check` framework will keep running all the rest of the tests in this directory, even if one of them fails.

If you don't put something like `set -euo pipefail` in your script, then you need to check the exit status of (nearly?) every command, which you aren't doing.  Otherwise, the test will fail inside this script, but then `make check` won't recognize it as failed, and the final count of past/fail/skip at the end of this directory will be inaccurate (i.e., silent test fails).

All of these options are so useful in catching errors that I generally `set -euo pipefail` in all my bash scripts these days...

Reply:
We did not use this script in `make check`. I pipe the result through grep to see how many tests failed, as the script test all MPI_Op for all types of predefined datatypes.

Reply:
Ah, I see.

You still might want to `set -u` to catch unbound variables.  Very helpful at preventing/finding bugs.